frontend/workflows/projectSelector/src/selector-reducer.tsx

```diff
@@ -1,9 +1,28 @@
 import type { clutch as IClutch } from "@clutch-sh/api";
 import _ from "lodash";
 
-import type { Action, State } from "./hello-world";
+import type { Action, ProjectState, State } from "./hello-world";
 import { deriveSwitchStatus, Group } from "./hello-world";
 
+const updateGroupstate = (
+  state: State,
+  group: Group,
+  project: string,
+  projectState: ProjectState
+): State => {
+  const newState = { ...state };
+  if (project in newState[group]) {
+    // preserve the checked value if the project is already in the group
+    newState[group][project].checked = state[group][project].checked;
```

Comment:
redundant
```suggestion
    newState[group][project].checked = state[group][project].checked;
```

Reply:
the suggested change is the same. was trying to figure out what you meant but couldn't spot the redundancy. going to merge this in to incorporate the helper in `add_projects` and can fix in the other PR!